### DISCUSSION TITLE
[Answered] Is there a way to have ddterm show in the alt+tab list when open?

### DISCUSSION THREAD

**Discussion by tledford4** [ddterm/gnome-shell-extension-ddterm#122] [Q&A] [answered]
95% of my navigation (application switching) is done via Alt+Tab or Alt+<Super>+Tab.  I suppose because it is an extension rather than a "standard" executable, ddterm isn't present in the list of applications that appears when pressing Alt+Tab.

Is there anything that can be done to expose ddterm and include it in the Alt+Tab list of running applications?

Hope this is clear, if not, let me know and I'll try to grab a screenshot.

**Comment by tledford4** [comment]
Never mind, I found the setting, it just wasn't labeled in such a way that it was clear to me that it would do what I wanted.

**Comment by tledford4** [comment, marked as answer]
"Exclude from overview and task bar" is selected by default, uncheck it for normal behavio
![Screenshot from 2022-03-07 09-55-19](https://user-images.githubusercontent.com/5845674/157058743-345600d5-f57a-4e30-a1c0-e8f656cf48cb.png)
r

**Comment by bogardon** [comment]
it only shows up in the alttab list when its in focus though, how do i get it to show up all the time?

EDIT: To be clear, my goal is to be able to alt tab between ddterm and somthing else, as well as being able to switch to it with a global hotkey.

**Reply by amezin** [reply]
Sorry, overlooked your comment somehow.

I'm not sure what exactly you mean by "its in focus". If "Above all windows" and "Hide from overview/task bar" are both disabled, ddterm window is present in alt-tab list, even if some other window is active. However, if ddterm window is not shown (i.e. "toggled off", by the keyboard shortcut or the icon on the panel), it's not present in the list - but it's expected.

**Reply by bogardon** [reply]
I see, so if ddterm is toggled off with the global hot key, it won't be in the alt-tab list. That makes sense, but I frequently make this mistake:

Say I begin from ddterm and alt-tab into firefox. I can alt-tab back to ddterm just fine, but if I attempt to use the global hot key, it won't work the first time because it actually hides ddterm. I have to press the global hot key a second time to bring it back. Not sure if I'm explaining that right.

Or say I begin from ddterm and use the global hotkey to hide it so that firefox is now in focus. Pressing alt-tab now will not return me to ddterm, but instead will take me to another application I did not expect.

My behavior is that I'll use alt-tab if I believe ddterm was the previously focused application. If I'm not sure, I'll use the global hot key. The fact that ddterm only shows up in the alt-tab list when it's toggled on throws me off, because now I need to remember if it's in the toggled on state or not.

**Reply by bogardon** [reply]
I could use the "Show/Activate Terminal Window" shortcut, but I can't use that same shortcut to hide it. Can this be a toggle too?

**Reply by amezin** [reply]
> Can this be a toggle too?

How should it behave exactly?

**Reply by bogardon** [reply]
what i tried was:

- disable exclude from overview/taskbar. this allows me to alt-tab to ddterm like it was a normal app.
- delete global shortcut for toggle terminal window. this prevents ddterm from being removed from the alt-tab list as long as it is running.
- add a global shortcut for show/activate terminal window. 

in this setup i can always use the new global shortcut to immediately bring ddterm into focus, no matter how many applications were in front of it in the alt-tab list. I can also alt-tab to ddterm if I wished, but the global shortcut is faster sometimes. This is almost perfect, except I cannot hide ddterm (in effect alt-tab to previous app) with the same global shortcut.

Is this making any sense? :)

**Reply by amezin** [reply]
> This is almost perfect, except I cannot hide ddterm (in effect alt-tab to previous app) with the same global shortcut.

And you're asking to make the same shortcut also hide ddterm... sometimes. But you're avoiding the tricky part - when exactly is that "sometimes"?

Also, you already can make ddterm hide automatically when you alt-tab to other app.

**Reply by bogardon** [reply]
Couldn't you just show if it's not in focus and hide if it's in focus? It would behave the same as the toggle terminal window shortcut, but wouldn't have the side effect of removing it from the alt-tab list when hiding.
